src/PowerShellEditorServices/Services/Workspace/WorkspaceService.cs

```diff
@@ -451,6 +450,22 @@ bool ignoreReparsePoints
             }
         }
 
+        internal static StreamReader OpenStreamReader(DocumentUri uri)
+        {
+            FileStream fileStream = new(uri.GetFileSystemPath(), FileMode.Open, FileAccess.Read);
+            // Default to UTF8 no BOM if a BOM is not present. Note that `Encoding.UTF8` is *with*
+            // BOM, so we call the ctor here to get the BOM-less version.
+            //
+            // TODO: Honor workspace encoding settings for the fallback.
+            return new StreamReader(fileStream, new UTF8Encoding(), detectEncodingFromByteOrderMarks: true);
```

Comment:
This is another thing I changed while I was in here. We were defaulting to UTF8 *with BOM* which seems wrong. We can tell it to use the BOM if it's available, and if not then fallback to UTF8 no-BOM.

Reply:
Though ofc ideally we'd be asking the client for encoding preferences.

Reply:
LGTM.